flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/resources/Resource.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.kubeclient.resources;
+
+import org.apache.flink.configuration.Configuration;
+
+/**
+ * Represent a kubernetes resource.
+ */
+public abstract class Resource<T> {
```

Comment:
Could we name it to be more specific like `KubernetesResource`?
We already have a class named `Resource` in Flink and I think a different name would be better for future code maintenance.

Reply:
In Flink project, `KubernetesResource` make more sense than `Resource`. It is a common name too much. I will rename `Resource` to `KubernetesResource`.